### DISCUSSION TITLE
High chance of kitty hanging up after a display is removed

### DISCUSSION THREAD

**Discussion by evaninf** [kovidgoyal/kitty#8159] [General]
I don't make a bug report because I'm not sure how to debug this: when I disconnect and reconnect a display or when there's a long period of screen inactivity and blanking there's a chance that the main kitty process (`kitty -1`) becomes frozen, all terminals turn black and the only way out is killing them.

I tried running with `gdb` and saving a core file when this happens but these files end up being over 5GiB each time.

This happens fairly frequently, some times up to four times per day, so what should I do and report when this happens again?

**Comment by kovidgoyal** [comment]
This will be an issue in GPU drivers and/or GLFW windowing code, not
kitty code. In order to debug it, you will first need to find a
reproducer otherwise it is too hard as there are too many variables
involved. You can build kitty with make debug-event-loop and run it with
--debug-rendering in addition both of these will get you more logs.

**Comment by evaninf** [comment]
Thanks for the suggestion, I'll try buiding and see what happens.

By the way, when the freeze happens only the windows belonging to a single instance are affected, terminals launched standalone aren't affected.

**Comment by kovidgoyal** [comment]
On Tue, Dec 24, 2024 at 08:02:22AM -0800, Evan wrote:
> Thanks for the suggestion, I'll try buiding and see what happens.
>
> By the way, when the freeze happens only the windows belonging to a single instance are affected, terminals launched standalone aren't affected.

Yes it's likely something to do with OpenGL contexts. But to debug we
really need to be able to repro.
